cinema/views.py

```
@@ -97,12 +99,28 @@
     def get_serializer_class(self):
         if self.action == "list":
             return MovieListSerializer
-
-        if self.action == "retrieve":
+        elif self.action == "retrieve":
             return MovieDetailSerializer
+        elif self.action == "upload_image":
+            return MovieImageSerializer
 
         return MovieSerializer
 
+    @action(
+        methods=["POST"],
+        detail=True,
+        permission_classes=[IsAdminUser],
+        url_path="upload-image",
+    )
+    def upload_image(self, request, pk=None):
+        movie = self.get_object()
+        serializer = self.get_serializer(movie, data=request.data)
+        if serializer.is_valid():
```

Use serializer.is_valid(raise_exception=True) to surface validation errors as standard DRF responses instead of manually checking and returning errors. Change:

if serializer.is_valid():
    serializer.save()
    return Response(...)

to:

serializer.is_valid(raise_exception=True)
serializer.save()
return Response(serializer.data, status=status.HTTP_200_OK)

Rationale: This simplifies the code and ensures DRF returns a proper 400 with details when validation fails for the uploaded file (e.g., invalid image). This also makes the endpoint behavior predictable in tests/clients.